app/preferences.go

```diff
@@ -50,15 +50,19 @@ func (p *preferences) forceImmediateSave() {
 func (p *preferences) resetSavedRecently() {
 	go func() {
 		time.Sleep(time.Millisecond * 100) // writes are not always atomic. 10ms worked, 100 is safer.
```

Comment:
I saw the message earlier but this still makes me kind of worried. Are we sure that the previous reported bug about writes not being atomic just weren't a race? Are writes not atomic or are we just missing locking between this and the place where it is written?

Reply:
In Go, calling `.Write()` on a file returns how many bytes it has written so it must be blocking until the write is complete. If this code and the write is happening on the same thread, we should not see any problem as far as I can tell.

As long as we are not reading and writing from different threads, I don't think atomic writes should matter much.

Reply:
My understanding was that it was at the operating system level - writing a file and immediately re-opening it for read is not guaranteed.
In my experience closing file handles has all sorts of challenges. I would rather explore this by way of re-examining the original issue in our new thread world rather than backing out the fix in this otherwise massive "enable race checking" PR.

Reply:
Absolutely. Sure